### PR TITLE
Issue 10352 ng site browser

### PR DESCRIPTION
**Pull request by jtesser**

### CODE REVIEW

File: src/dotcms-js/core/util/filesystem.service.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        private log: LoggerService
+    ) {}
+
+    /**
+     * Utility method to help crawl a local directoy returning all folders and files underneath it
+     * @param directory Local Filesystem Directory to crawl
+     * @param files Can be NULL. Is a list of Strings to local File paths that will be returned.
+     * @returns {String[]} List of File and Folder Paths under the passed in path added to the passed in files[]
+     */
+    recurseDirectory(directory: string, files: String[]): String[] {
```
Comment (fmontes) [inline src/dotcms-js/core/util/filesystem.service.ts]: Are you using this method?

File: src/dotcms-js/core/util/filesystem.service.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        // };
+        return files;
+    }
+
+    /**
+     * Tests if a local file path is a directory or not
+     * @param localPath local file or directory path
+     * @returns {boolean}
+     */
+    isDirectory(localPath: string): boolean {
```
Comment (fmontes) [inline src/dotcms-js/core/util/filesystem.service.ts]: Are you using this method?

File: src/dotcms-js/core/util/http.service.ts [hunk 3]
```diff
@@ -0,0 +1,135 @@
+import {Injectable} from '@angular/core';
+import {Http, Headers, Response, RequestMethod, RequestOptions} from '@angular/http';
+import {Observable} from 'rxjs';
+
+import 'rxjs/add/operator/map';
```
Comment (fmontes) [inline src/dotcms-js/core/util/http.service.ts]: I don't think you need to import this... we use it in other places without importing them.

File: src/dotcms-js/core/util/notification.service.ts [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    iconPath: string;
+
+    constructor
+    (
+        private config: AppConfig
+    ) {
+        this.iconPath = config.iconPath;
+    }
+
+    displayErrorMessage(body: string): void {
```
Comment (fmontes) [inline src/dotcms-js/core/util/notification.service.ts]: Missin doc

File: src/dotcms-js/components/site-datatable/site-datatable.component.ts [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                    this.loadSite(siteName);
+                }
+            });
+        this.subscription = updateService.currentURI
+            .subscribe(uri => {
+                if (uri) {
+                    this.loadFolder(uri);
+                }
+            });
+        setTimeout(() => {
```
Comment (jtesser) [inline src/dotcms-js/components/site-datatable/site-datatable.component.ts]: Timeout fixed a browser issue.  Is silly but was needed

File: src/dotcms-js/components/site-datatable/site-datatable.component.ts [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+     * @param siteName
+     */
+    loadSite(siteName: string): void {
+        this.siteName = siteName;
+        this.siteBrowserService.getTreeableAssetsUnderSite(siteName)
+            .subscribe((treeables: Treeable[]) => this.treeables = treeables);
+        setTimeout(() => {}, 100);
+    }
+
+    handleDragOver(e: any): void {
```
Comment (jtesser) [inline src/dotcms-js/components/site-datatable/site-datatable.component.ts]: yes

File: src/dotcms-js/components/treeable-detail/treeable-detail.component.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        if (updateService.getSelectedTreeable()) {this.treeable = updateService.getSelectedTreeable(); }
+        this.subscription = updateService.currentTreeable
+            .subscribe(treeable => {
+                if (treeable) {
+                    this.treeable = treeable;
+                }else {
+                    this.treeable = new Treeable();
+                }
+            });
+        setTimeout(() => {
```
Comment (jtesser) [inline src/dotcms-js/components/treeable-detail/treeable-detail.component.ts]: fixed a race condition issue

**Pull request merged**